File: frontend/src/app/core/model/SeatSearchCriteria.ts
import {SeatPlace} from '../enums/SeatPlace';
import {LegSpace} from '../enums/LegSpace';
import {SeatClass} from '../enums/SeatClass';

export interface SeatSearchCriteria {
  flightId: number
  place: SeatPlace
  legSpace: LegSpace
  seatClass: SeatClass
}

File: backend/src/main/java/ee/assignment/backend/criteria/SeatSearchCriteria.java
package ee.assignment.backend.criteria;

import ee.assignment.backend.enums.LegSpace;
import ee.assignment.backend.enums.SeatClass;
import ee.assignment.backend.enums.SeatPlace;
import lombok.Data;

@Data
public class SeatSearchCriteria {
    private Long flightId;
    private SeatPlace place;
    private LegSpace legSpace;
    private SeatClass seatClass;
    private Boolean nextToExit;
}
